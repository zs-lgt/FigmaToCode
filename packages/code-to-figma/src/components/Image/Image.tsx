import * as React from 'react';
import { RectangleProps, Rectangle } from '../rectangle/Rectangle';
import { ResizeMode } from '../../styleTransformers/transformGeometryStyleProperties';
import { StyleSheet } from '../..';

export interface ImageProps extends RectangleProps {
    source: string | { uri: string } | { default: string };
    resizeMode?: ResizeMode;
}

const Image: React.FC<ImageProps> = props => {
    const { style, source, resizeMode } = props;
    return (
        <Rectangle
            {...props}
            style={[StyleSheet.flatten(style), { backgroundImage: source, backgroundSize: resizeMode }]}
        />
    );
};

export { Image };
