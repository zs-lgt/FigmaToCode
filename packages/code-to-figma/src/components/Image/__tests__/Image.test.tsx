import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Image } from '../Image';
import { createFigma } from 'figma-api-stub';

describe('<Image />', () => {
    beforeEach(() => {
        // @ts-ignore
        global.figma = createFigma({
            simulateErrors: true
        });
    });

    it('Image with source and layout style properties', () => {
        const tree = renderer
            .create(
                <Image
                    source="./image.png"
                    style={{
                        width: 100,
                        height: 200
                    }}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Image with resizeMode prop', () => {
        const tree = renderer
            .create(
                <Image
                    source="./image.png"
                    resizeMode="stretch"
                    style={{
                        width: 100,
                        height: 200
                    }}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
