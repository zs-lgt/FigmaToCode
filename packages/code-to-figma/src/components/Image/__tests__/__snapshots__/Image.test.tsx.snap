// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> Image with resizeMode prop 1`] = `
<rectangle
  height={200}
  innerRef={
    Object {
      "current": undefined,
    }
  }
  resizeMode="stretch"
  source="./image.png"
  style={
    Object {
      "backgroundImage": "./image.png",
      "backgroundSize": "stretch",
      "height": 200,
      "width": 100,
    }
  }
  width={100}
/>
`;

exports[`<Image /> Image with source and layout style properties 1`] = `
<rectangle
  height={200}
  innerRef={
    Object {
      "current": undefined,
    }
  }
  source="./image.png"
  style={
    Object {
      "backgroundImage": "./image.png",
      "backgroundSize": undefined,
      "height": 200,
      "width": 100,
    }
  }
  width={100}
/>
`;
